In [15]:
import pandas as pd 
import numpy as np

pd.options.display.float_format = '{:,.2f}'.format

## 1. Carga de los dataset

Se realiza la carga de los dos datasets con los que se va a trabajar y se realiza un breve análisis de su estructura y contenido

In [7]:
credit_df = pd.read_csv("../data/credit_record.csv")

In [8]:
credit_df

ID  MONTHS_BALANCE STATUS
0        5001711               0      X
1        5001711              -1      0
2        5001711              -2      0
3        5001711              -3      0
4        5001712               0      C
...          ...             ...    ...
1048570  5150487             -25      C
1048571  5150487             -26      C
1048572  5150487             -27      C
1048573  5150487             -28      C
1048574  5150487             -29      C

[1048575 rows x 3 columns]

In [9]:
application_df = pd.read_csv('../data/application_record.csv')

In [10]:
application_df

ID CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  CNT_CHILDREN  \
0       5008804           M            Y               Y             0   
1       5008805           M            Y               Y             0   
2       5008806           M            Y               Y             0   
3       5008808           F            N               Y             0   
4       5008809           F            N               Y             0   
...         ...         ...          ...             ...           ...   
438552  6840104           M            N               Y             0   
438553  6840222           F            N               N             0   
438554  6841878           F            N               N             0   
438555  6842765           F            N               Y             0   
438556  6842885           F            N               Y             0   

        AMT_INCOME_TOTAL      NAME_INCOME_TYPE            NAME_EDUCATION_TYPE  \
0               427500.0               Working               Higher education   
1               427500.0               Working               Higher education   
2               112500.0               Working  Secondary / secondary special   
3               270000.0  Commercial associate  Secondary / secondary special   
4               270000.0  Commercial associate  Secondary / secondary special   
...                  ...                   ...                            ...   
438552          135000.0             Pensioner  Secondary / secondary special   
438553          103500.0               Working  Secondary / secondary special   
438554           54000.0  Commercial associate               Higher education   
438555           72000.0             Pensioner  Secondary / secondary special   
438556          121500.0               Working  Secondary / secondary special   

          NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  DAYS_BIRTH  DAYS_EMPLOYED  \
0             Civil marriage   Rented apartment      -12005          -4542   
1             Civil marriage   Rented apartment      -12005          -4542   
2                    Married  House / apartment      -21474          -1134   
3       Single / not married  House / apartment      -19110          -3051   
4       Single / not married  House / apartment      -19110          -3051   
...                      ...                ...         ...            ...   
438552             Separated  House / apartment      -22717         365243   
438553  Single / not married  House / apartment      -15939          -3007   
438554  Single / not married       With parents       -8169           -372   
438555               Married  House / apartment      -21673         365243   
438556               Married  House / apartment      -18858          -1201   

        FLAG_MOBIL  FLAG_WORK_PHONE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  \
0                1                1           0           0             NaN   
1                1                1           0           0             NaN   
2                1                0           0           0  Security staff   
3                1                0           1           1     Sales staff   
4                1                0           1           1     Sales staff   
...            ...              ...         ...         ...             ...   
438552           1                0           0           0             NaN   
438553           1                0           0           0        Laborers   
438554           1                1           0           0     Sales staff   
438555           1                0           0           0             NaN   
438556           1                0           1           0     Sales staff   

        CNT_FAM_MEMBERS  
0                   2.0  
1                   2.0  
2                   2.0  
3                   1.0  
4                   1.0  
...                 ...  
438552              1.0  
438553              1.0  
438554              1.0  
43855

## 2. Análisis de los datasets

In [47]:
len(application_df)

438557

In [46]:
# Descrición de las variables numéricas
application_df[['AMT_INCOME_TOTAL','CNT_FAM_MEMBERS','CNT_CHILDREN']].describe()

AMT_INCOME_TOTAL  CNT_FAM_MEMBERS  CNT_CHILDREN
count        438,557.00       438,557.00    438,557.00
mean         187,524.29             2.19          0.43
std          110,086.85             0.90          0.72
min           26,100.00             1.00          0.00
25%          121,500.00             2.00          0.00
50%          160,780.50             2.00          0.00
75%          225,000.00             3.00          1.00
max        6,750,000.00            20.00         19.00

In [43]:
application_df.isna().sum()

ID                     0
CODE_GENDER            0
FLAG_OWN_CAR           0
FLAG_OWN_REALTY        0
CNT_CHILDREN           0
AMT_INCOME_TOTAL       0
NAME_INCOME_TYPE       0
NAME_EDUCATION_TYPE    0
NAME_FAMILY_STATUS     0
NAME_HOUSING_TYPE      0
DAYS_BIRTH             0
DAYS_EMPLOYED          0
FLAG_MOBIL             0
FLAG_WORK_PHONE        0
FLAG_PHONE             0
FLAG_EMAIL             0
CNT_FAM_MEMBERS        0
dtype: int64

El dataset *application_df* cuenta con un total de 438557 observaciones. Únicamente la variable OCCUPATION_TYPE cuenta con un gran número de valores nulos, equivalente al 30.6% del total de registros. Por lo que esta variable no será tomada en cuenta para el presente proyecto, ya que el procentaje de valores nulos es muy alto para rellenarlo utilizando algún método. 

In [28]:
application_df = application_df.drop(columns=['OCCUPATION_TYPE'])

In [29]:
# Análisis del tipo de datos
application_df.dtypes

ID                       int64
CODE_GENDER             object
FLAG_OWN_CAR            object
FLAG_OWN_REALTY         object
CNT_CHILDREN             int64
AMT_INCOME_TOTAL       float64
NAME_INCOME_TYPE        object
NAME_EDUCATION_TYPE     object
NAME_FAMILY_STATUS      object
NAME_HOUSING_TYPE       object
DAYS_BIRTH               int64
DAYS_EMPLOYED            int64
FLAG_MOBIL               int64
FLAG_WORK_PHONE          int64
FLAG_PHONE               int64
FLAG_EMAIL               int64
CNT_FAM_MEMBERS        float64
dtype: object

In [40]:
# Verificamos el tipo de dato que debería tener cada variable
application_df.convert_dtypes().dtypes

ID                       Int64
CODE_GENDER             string
FLAG_OWN_CAR            string
FLAG_OWN_REALTY         string
CNT_CHILDREN             Int64
AMT_INCOME_TOTAL       float64
NAME_INCOME_TYPE        string
NAME_EDUCATION_TYPE     string
NAME_FAMILY_STATUS      string
NAME_HOUSING_TYPE       string
DAYS_BIRTH               Int64
DAYS_EMPLOYED            Int64
FLAG_MOBIL               Int64
FLAG_WORK_PHONE          Int64
FLAG_PHONE               Int64
FLAG_EMAIL               Int64
CNT_FAM_MEMBERS          Int64
dtype: object